src/training/validator.h

```diff
@@ -180,6 +177,9 @@ class CrossEntropyValidator : public Validator<data::Corpus> {
       }
     }
 
+    // get back to the original cost type
+    options_->set("cost-type", ctype);
+
```

Comment:
As said above. Modification should be safer. This seems too fragile.

Reply:
See comment above.